neotron-bmc-pico/src/main.rs

```diff
@@ -776,6 +845,59 @@ where
 				proto::Response::new_without_data(proto::ResponseResult::BadLength)
 			}
 		}
+		(proto::RequestType::Read, Ok(Command::SpeakerDuration)) => {
+			defmt::debug!("Reading speaker duration");
+			proto::Response::new_ok_with_data(unsafe {
+				&any_as_u8_slice(&register_state.speaker_config)[0..1]
+			})
+		}
+		(proto::RequestType::ShortWrite, Ok(Command::SpeakerDuration)) => {
+			defmt::debug!("Writing speaker duration");
+			register_state
+				.speaker_config
+				.set_duration(req.length_or_data);
+			register_state.speaker_needs_update = true;
+			proto::Response::new_without_data(proto::ResponseResult::Ok)
+		}
+		(proto::RequestType::Read, Ok(Command::SpeakerPeriodHigh)) => {
+			defmt::debug!("Reading speaker period (high)");
+			proto::Response::new_ok_with_data(unsafe {
+				&any_as_u8_slice(&register_state.speaker_config)[1..2]
+			})
+		}
+		(proto::RequestType::ShortWrite, Ok(Command::SpeakerPeriodHigh)) => {
+			defmt::debug!("Writing speaker period (high)");
+			register_state.speaker_config.set_ticks(
+				(register_state.speaker_config.ticks() & 0xff) | ((req.length_or_data as u16) << 8),
```

Comment:
Maybe there should be a method called `set_ticks_high()` which does this internally.